app/src/models/repository.ts

```diff
@@ -59,6 +59,7 @@ export class Repository implements IRepository {
   }
 
   public get name(): string {
-    return path.basename(this.path)
+    const repoName = this.gitHubRepository && this.gitHubRepository.name
```

Comment:
Given `this.path` and `this.githubRepository` are set in the constructor, and how much we use this, we should compute this there and cache it